pkg/source/imgpkg.go

```diff
@@ -64,8 +67,11 @@ func ImgpkgPush(ctx context.Context, dir string, excludedFiles []string, registr
 		return "", fmt.Errorf("parsing '%s': %s", image, err)
 	}
 
+	levelLogger := util.NewUILevelLogger(util.LogWarn, ui.NewWriterUI(c.Stdout, c.Stderr, nil))
+	imagesUploaderLogger := util.NewProgressBar(levelLogger, "", "Error uploading images")
+	regWithProgress := registry.NewRegistryWithProgress(reg, imagesUploaderLogger)
 	excludedFiles = append(excludedFiles, path.Join(dir, ".imgpkg"))
-	digest, err := plainimage.NewContents([]string{dir}, excludedFiles).Push(uploadRef, nil, reg, ui.NewNoopUI())
+	digest, err := plainimage.NewContents([]string{dir}, excludedFiles).Push(uploadRef, nil, regWithProgress, ui.NewNoopUI())
```

Comment:
There is an existing patterns to pass reg via context so why wouldnt you do that instead of passing cli config in this func?

Reply:
Updated the function to separate the registry creation logic with the imgpkg push